client/my-sites/stats/site.jsx

```diff
@@ -267,23 +265,75 @@ class StatsSite extends Component {
 					</div>
 				</div>
 				<JetpackColophon />
+			</>
+		);
+	}
+
+	renderEnableStatsModule() {
+		const { slug, siteId, path } = this.props;
+		return (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ translate( 'Looking for Stats?' ) }
+				line={ translate(
+					'Enable Site Stats below or visit {{a}}Traffic > Site Stats{{/a}} for more options',
+					{
+						components: {
+							a: <a href={ `/marketing/traffic/${ slug }` } />,
+						},
+					}
+				) }
+				action={ translate( 'Enable Site Stats' ) }
+				actionCallback={ () => this.props.enableJetpackStatsModule( siteId, path ) }
+			/>
+		);
+	}
+
+	render() {
+		const { isJetpack, siteId, showEnableStatsModule } = this.props;
+
+		const { period } = this.props.period;
+
+		return (
+			<Main wideLayout={ true }>
+				<QueryKeyringConnections />
+				{ isJetpack && siteId && <QueryJetpackModules siteId={ siteId } /> }
+				{ siteId && <QuerySiteKeyrings siteId={ siteId } /> }
+				<DocumentHead title={ translate( 'Stats and Insights' ) } />
+				<PageViewTracker
+					path={ `/stats/${ period }/:site` }
+					title={ `Stats > ${ titlecase( period ) }` }
+				/>
+				{ showEnableStatsModule ? this.renderEnableStatsModule() : this.renderStats() }
 			</Main>
 		);
 	}
 }
+const enableJetpackStatsModule = ( siteId, path ) =>
```

Comment:
Is there any benefit of declaring this function outside of the component? I have no strong feelings towards moving it, but was just curious.

Reply:
No special reason. If this turns into simple function later, then it'd be outside of the component. I don't have a strong preference here, for example I can move this all in mapDispatchToProps if I pick the longform syntax. Though this does force other folks to also use this for other actions when they don't need prop info.

```jsx
const mapDispatchToProps = ( dispatch, ownProps ) => {
  return {
    enableJetpackStatsModule: () => dispatch( withAnalytics(
		recordTracksEvent( 'calypso_jetpack_module_toggle', {
			module: 'stats',
			ownProps.path,
			toggled: 'on',
		} ),
		activateModule( ownProps.siteId, 'stats' )
	) );
  }
}
```